src/transforms/onehot.jl

```diff
@@ -0,0 +1,84 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+"""
+    OneHot(col)
+    
+Transforms categorical column `col` into one-hot columns of levels
+returned by the `levels` function of CategoricalArrays.jl.
+
+# Examples
+
+```julia
+OneHot(1)
+OneHot(:a)
+OneHot("a")
+```
+"""
+struct OneHot{S<:ColSelector} <: Stateless
+  col::S
+end
+
+isrevertible(::Type{<:OneHot}) = true
+
+_colname(col::Integer, names) = names[col]
+_colname(col::AbstractString, names) =
+  _colname(Symbol(col), names)
+
+function _colname(col::Symbol, names)
+  @assert col ∈ names "Invalid column selection."
+  return col
+end
```

Comment:
So you are reproducing the logic of the ColSpec feature here right? Maybe we can reuse the ColSpec functionality but restrict at construction time what ColSpec are allowed? In this case we want to restrict ourselves to a ColSpec that consist of a single column, right?